spec/features/warehouse_messaging_spec.rb

```diff
@@ -0,0 +1,61 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe 'Warehouse Messaging', type: :feature do
+  let(:num_plates) { 10 }
+  let!(:sci_swipe_card_id) { generate(:swipe_card_id) }
+  let!(:scientist) { create(:scientist, swipe_card_id: sci_swipe_card_id) }
+
+  def testing_scenario
+    location = create(:location_with_parent)
+    labwares = build_list(:labware, num_plates)
+    visit new_scan_path
+    expect do
+      fill_in "User swipe card id/barcode", with: sci_swipe_card_id
+      fill_in "Location barcode", with: location.barcode
+      fill_in "Labware barcodes", with: labwares.join_barcodes
+      click_button "Go!"
+    end.to change(Scan, :count).by(1)
+  end
+
+  context 'when using RabbitMQ' do
+    before do
+      Rails.configuration.bunny['enabled'] = true
+
```

Comment:
This is mutating the global configuration, and will affect subsequent tests.  Rather than doing this, you might be better off stubbing out the Handle constant with a new handler, or even an outright instance_double as you're stubbing the methods anyway.

https://relishapp.com/rspec/rspec-mocks/v/2-11/docs/stubbing-constants

Reply:
Would be something like this (not tested, so almost certainly errors)
```ruby
let(:broker) { instance_double(Messages::Broker) }
before do
  stub_const("Broker::Handle", broker)
end
```

Then all cases of `allow_any_instance_of(Messages::Broker)` can just become `allow(broker)`

Reply:
Oh, it looks like the tests are actually trying to connect to rabbitMQ, and then failing because the port isn't specified. So you're testing a little more of the broker here than I first thought. If you're looking on this as a full integration test, then perhaps create an actual real broker, but mock the bunny connections?

Reply:
This should be done in the `spec/messages/broker_spec.rb`